src/components/interpolatedColorLegend.ts

```diff
@@ -221,56 +222,61 @@ export module Components {
 
       let boundingBoxAttr: { [key: string]: number } = {
         x: 0,
-        y: padding,
+        y: 0,
         width: 0,
         height: 0
       };
 
+      let padding: number;
+
       let numSwatches = InterpolatedColorLegend._DEFAULT_NUM_SWATCHES;
-      let textHeight = this._measurer.measure().height;
+
       if (this.expands() && textHeight > 0) {
-        let offset = this._isVertical() ? 2 * padding :  4 * padding - text0Width - text1Width;
+        let offset = this._isVertical() ? 0 :  2 * textPadding - text0Width - text1Width;
         let fullLength = this._isVertical() ? this.height() : this.width();
         numSwatches = Math.max(Math.floor((fullLength - offset) / textHeight), numSwatches);
       }
 
       if (this._isVertical()) {
+        padding = (this.width() - Math.max(text0Width, text1Width) - 2 * this._textPadding) / 2;
         let longestTextWidth = Math.max(text0Width, text1Width);
```

Comment:
Let's move this calculation one line up and then use this variable in setting `padding`